### ISSUE TITLE
Integrating Tracing Support for HAProxy in Instana

### ISSUE THREAD

**Issue by vishnusemirc** [haproxy/haproxy#2782]
### Your Feature Request

Instana is planning to add tracing support to HAProxy and needs to confirm a few details before proceeding. We've seen in Git Hub that HAProxy is planning to integrate the OpenTelemetry (OTEL) support, so we'd like to confirm the current status of OTEL support.
Also we would like to understand if there's an opportunity for us to contribute our Instana tracing plugin, which is based on OpenTracing, while ensuring long-term support.
Could you please confirm which method would be preferable for you—OpenTracing or OpenTelemetry?
If the OTEL support is feasible, we'd like to know if we can contribute specific code changes, such as dropping Instana headers, adding resource attributes, infrastructure linking, OTel metrics.

Your feedback will help us to plan the best way to proceed.

### What are you trying to do?

Planning to add tracing support to HAProxy in Instana.

### Output of `haproxy -vv`

```plain
NA
```

**Comment by vishnusemirc** [comment]
Hi @capflam,

Could you please help me on this?
TIA.

**Comment by capflam** [comment]
I don't know how Instana works, so it is hard for me to fully answer you.

About the OpenTracing support. Because the project is fully dead since now 2 years, the HAProxy add-on will be removed, may be after a deprecation period, but  independently  on OpenTelemetry support. An OpenTelemetry add-on will be welcome if anyone is willing to work on it and to maintain it. I'm tempted to say that if it was not implemented in 2 years to replace the Opentracing add-on, it means there is not a strong traction from the community. But I may be wrong.

Technically speaking, you must base your work on OpenTelemetry, definitely. But I don't know what implies an Instana tracing plugin. You must elaborate. But first of all, you should land the subject on the mailing-list because add-ons are only added if there are someone ready to use it. It is useless to have a long list of unused add-on.

**Comment by capflam** [comment]
Of course, you must also check it is compatible with the HAproxy licence. You may check CONTRIBUTING file.

**Comment by zaga00** [comment]
The OpenTelemetry filter is under development and will be very similar to the current OpenTracing filter.  However, it is not that simple (as it was not simple for OpenTracing either) because first I have to make a C wrapper for opentelemetry-cpp API.  It will probably be usable in the 3.2 version of HAProxy.

Also, even though OpenTelemetry has been in development for several years, it has not been a complete and stable C++ API for a long time, so I waited for it to finish.

Regarding Instana, never worked with it, but I would repeat what @capflam wrote: "base it on OpenTelemetry".

**Comment by vishnusemirc** [comment]
@zaga00 and @capflam,

Thanks for the update.

Could you please confirm the roadmap for the upcoming HAProxy version 3.2, includes OTEL support?

**Comment by odoucet** [comment]
Hello, Any update on the subject ? Is it still planned for 3.2 ?

**Comment by zaga00** [comment]
I don't get to do it for version 3.2, because of other commitments.  First I need to make a C wrapper for OpenTelemetry C++ library, and then add OTEL filter to haproxy.  Both projects are in progress, but not ready for the public..

**Comment by wtarreau** [comment]
@zaga00 please keep everyone updated about your expectations / hopes so that we can update the deprecation status. OT is still supposed to be **deprecated in 3.3** and **removed in 3.5**, and users need to be warned upfront so that they can prepare their migration. Is it still OK for you if we mark it right now as deprecated in 3.3 so that new users don't jump into it and old ones start to watch for updates on the topic ? For reference the status is still mentioned there: [https://github.com/haproxy/wiki/wiki/Breaking-changes](https://github.com/haproxy/wiki/wiki/Breaking-changes). Thanks a lot!

**Comment by miroslavmasaryk3221** [comment]
Do we have rough date/version when OTEL tracing would be available? 

Thanks

**Comment by zaga00** [comment]
Hello all, OpenTelemetry will be supported in HAProxy very soon.

**Comment by zaga00** [comment]
The first step is done - I published the OpenTelemetry C Wrapper library code: https://github.com/haproxytech/opentelemetry-c-wrapper

**Comment by zaga00** [comment]
The OpenTelemetry filter has been added to the HAProxy source code.

**Comment by wtarreau** [comment]
Should we just close this issues since it's already addressed in #1640 ?
